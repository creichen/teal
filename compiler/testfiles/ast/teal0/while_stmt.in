fun f(a : int, b) : int = {
    while a {
	  b := b + 1;
	  a := a - 1;
    }
    return b;
}