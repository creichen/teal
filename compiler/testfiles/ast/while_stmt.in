fun f(a : T, b) : U = {
    while a {
	  b := b + 1;
	  a := a - 1;
    }
    return b;
}