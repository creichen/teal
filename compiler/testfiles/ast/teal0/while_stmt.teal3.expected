Module
  List
  List
    FunDecl
      IdDecl Identifier="f"
      Opt
        Type
          IdUse Identifier="int"
          List
          List
      List
        VarDecl
          IdDecl Identifier="a"
          Opt
            Type
              IdUse Identifier="int"
              List
              List
          Opt
        VarDecl
          IdDecl Identifier="b"
          Opt
          Opt
      Opt
        BlockStmt
          List
            WhileStmt
              Access
                IdUse Identifier="a"
              BlockStmt
                List
                  AssignStmt
                    Access
                      IdUse Identifier="b"
                    AddExpr
                      Access
                        IdUse Identifier="b"
                      IntConstant Value="1"
                  AssignStmt
                    Access
                      IdUse Identifier="a"
                    SubExpr
                      Access
                        IdUse Identifier="a"
                      IntConstant Value="1"
            ReturnStmt
              Access
                IdUse Identifier="b"
